Add support for vips 8.16.0

When trying to compile our application with vips 8.16.0 & govips 2.15.0 I get the following error:
```
# github.com/davidbyttow/govips/v2/vips
vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:23:43: could not determine kind of name for C.VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO
vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:26:43: could not determine kind of name for C.VIPS_FOREIGN_JPEG_SUBSAMPLE_LAST
vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:25:43: could not determine kind of name for C.VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF
vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:24:43: could not determine kind of name for C.VIPS_FOREIGN_JPEG_SUBSAMPLE_ON
vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:355:20: could not determine kind of name for C.VipsForeignJpegSubsample
cgo: 
clang errors for preamble:
In file included from vendor/github.com/davidbyttow/govips/v2/vips/foreign.go:3:
./foreign.h:87:3: error: unknown type name 'VipsForeignJpegSubsample'; did you mean 'VipsForeignSubsample'?
   87 |   VipsForeignJpegSubsample jpegSubsample;
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
      |   VipsForeignSubsample
/thumbs/include/vips/foreign.h:475:3: note: 'VipsForeignSubsample' declared here
  475 | } VipsForeignSubsample;
      |   ^
1 error generated.
```

Looks like the enums were removed in this commit: https://github.com/libvips/libvips/commit/a935ba7a7b197e96db532ceffcea0c8b430cba6b

Would be great if you could make your lib work with the newest vips version :slightly_smiling_face:

--- a/vips/foreign.c
+++ b/vips/foreign.c
@@ -526,7 +526,7 @@ static SaveParams defaultSaveParams = {
     .stripMetadata = FALSE,
 
     .jpegOptimizeCoding = FALSE,
-    .jpegSubsample = VIPS_FOREIGN_JPEG_SUBSAMPLE_ON,
+    .jpegSubsample = VIPS_FOREIGN_SUBSAMPLE_ON,
     .jpegTrellisQuant = FALSE,
     .jpegOvershootDeringing = FALSE,
     .jpegOptimizeScans = FALSE,

--- a/vips/foreign.go
+++ b/vips/foreign.go
@@ -19,10 +19,10 @@ type SubsampleMode int
 
 // SubsampleMode enum correlating to libvips subsample modes
 const (
-	VipsForeignSubsampleAuto SubsampleMode = C.VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO
-	VipsForeignSubsampleOn   SubsampleMode = C.VIPS_FOREIGN_JPEG_SUBSAMPLE_ON
-	VipsForeignSubsampleOff  SubsampleMode = C.VIPS_FOREIGN_JPEG_SUBSAMPLE_OFF
-	VipsForeignSubsampleLast SubsampleMode = C.VIPS_FOREIGN_JPEG_SUBSAMPLE_LAST
+	VipsForeignSubsampleAuto SubsampleMode = C.VIPS_FOREIGN_SUBSAMPLE_AUTO
+	VipsForeignSubsampleOn   SubsampleMode = C.VIPS_FOREIGN_SUBSAMPLE_ON
+	VipsForeignSubsampleOff  SubsampleMode = C.VIPS_FOREIGN_SUBSAMPLE_OFF
+	VipsForeignSubsampleLast SubsampleMode = C.VIPS_FOREIGN_SUBSAMPLE_LAST
 )
 
 // ImageType represents an image type
@@ -358,7 +358,7 @@ func vipsSaveJPEGToBuffer(in *C.VipsImage, params JpegExportParams) ([]byte, err
 	p.quality = C.int(params.Quality)
 	p.interlace = C.int(boolToInt(params.Interlace))
 	p.jpegOptimizeCoding = C.int(boolToInt(params.OptimizeCoding))
-	p.jpegSubsample = C.VipsForeignJpegSubsample(params.SubsampleMode)
+	p.jpegSubsample = C.VipsForeignSubsample(params.SubsampleMode)
 	p.jpegTrellisQuant = C.int(boolToInt(params.TrellisQuant))
 	p.jpegOvershootDeringing = C.int(boolToInt(params.OvershootDeringing))
 	p.jpegOptimizeScans = C.int(boolToInt(params.OptimizeScans))
@@ -463,7 +463,7 @@ func vipsSaveJP2KToBuffer(in *C.VipsImage, params Jp2kExportParams) ([]byte, err
 	p.jp2kLossless = C.int(boolToInt(params.Lossless))
 	p.jp2kTileWidth = C.int(params.TileWidth)
 	p.jp2kTileHeight = C.int(params.TileHeight)
-	p.jpegSubsample = C.VipsForeignJpegSubsample(params.SubsampleMode)
+	p.jpegSubsample = C.VipsForeignSubsample(params.SubsampleMode)
 
 	return vipsSaveToBuffer(p)
 }

--- a/vips/foreign.h
+++ b/vips/foreign.h
@@ -84,7 +84,7 @@ typedef struct SaveParams {
 
   // JPEG
   BOOL jpegOptimizeCoding;
-  VipsForeignJpegSubsample jpegSubsample;
+  VipsForeignSubsample jpegSubsample;
   BOOL jpegTrellisQuant;
   BOOL jpegOvershootDeringing;
   BOOL jpegOptimizeScans;

